### PR TITLE
Added missing dependencies for jsdocs

### PR DESCRIPTION
**Pull request by tshneour**
## Tracking Info

Resolves jsdocs workflow failing

## Changes

- Added missing dependencies for jsdocs
- Added config file to recursively search `source` for js files

## Testing

- I don't see a good way to test this without deploying to GitHub pages, needs review

## Confirmation of Change

Files changed/added in `/source`:
- `package.json` (changed)
- `package-lock.json` (changed)
- `jsdoc.conf.json` (added)

### CODE REVIEW

**Comment by advaithm22** [comment]
these changes look good, I think there may be an issue with the current jsdocs.yml format. I think should be accessing GITHUB_TOKEN field instead of DEPLOY_KEY. See https://github.com/ncsu-landscape-dynamics/pops-core/issues/106.

**Comment by advaithm22** [comment]
can you test this by creating a small test branch, merging this one into that and then launching the actions workflow?

**Comment by tshneour** [comment]
> can you test this by creating a small test branch, merging this one into that and then launching the actions workflow?

There was a method header that was incorrectly formatted. I just regenerated the deploy token and it works now. Jsdocs are on a separate branch `gh-pages`.

File: .github/workflows/jsdocs.yml
Comment (advaithm22) [inline .github/workflows/jsdocs.yml:1]: I assume this is just for testing purposes but make sure you change this back to just staging before it's merged

**Review by advaithm22 — APPROVED** [review]

**Pull request merged**